apps/platform-app/src/hooks/useWarnIfUnsavedChanges/useWarnIfUnsavedChanges.tsx

```diff
@@ -0,0 +1,63 @@
+import { useModal } from '@/contexts/ModalContext'
+import { type NavigateOptions } from 'next/dist/shared/lib/app-router-context.shared-runtime'
+import { useRouter } from 'next/navigation'
+import { useEffect } from 'react'
+import { ConfirmLeaveModal } from './ConfirmLeaveModal/ConfirmLeaveModal'
+
+//This hook, when it receives a value of true, will display a modal confirming the user's intention to leave before modifying the router.
+//If the user tries to leave the page using the browser interface (back button, refresh, close, etc.), a native confirm box will be displayed.
+//Other cases within the application may not be handled correctly.
+
+export const useWarnIfUnsavedChanges = (unsaved: boolean) => {
+  const router = useRouter()
+  const { closeModal, showModal } = useModal()
+
+  useEffect(() => {
+    const beforeUnloadHandler = (e: BeforeUnloadEvent) => {
+      e.preventDefault()
```

Comment:
**suggestion**: Consider adding a message to the `beforeUnloadHandler` to provide more context to the user when they attempt to leave the page.

Reply:
I use native message provides by browser